torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -2876,38 +2877,53 @@ at::Tensor scatter_reduce_helper(const at::Tensor& self, int64_t dim,
 
 at::Tensor XLANativeFunctions::scatter(const at::Tensor& self, int64_t dim,
                                        const at::Tensor& index,
-                                       const at::Tensor& src) {
+                                       const at::Tensor& src,
+                                       bool unique_indices) {
   XLA_FN_COUNTER("xla::");
+  XLA_CHECK(unique_indices)
+      << "scatter not have a deterministic implementation";
   return scatter_reduce_helper(self, dim, index, src, c10::nullopt);
 }
 
 at::Tensor XLANativeFunctions::scatter(const at::Tensor& self, int64_t dim,
                                        const at::Tensor& index,
-                                       const at::Scalar& value) {
+                                       const at::Scalar& value,
+                                       bool unique_indices) {
   XLA_FN_COUNTER("xla::");
+  XLA_CHECK(unique_indices)
+      << "scatter not have a deterministic implementation";
   return scatter_reduce_helper(self, dim, index, value, c10::nullopt);
 }
 
 at::Tensor XLANativeFunctions::scatter(const at::Tensor& self, int64_t dim,
                                        const at::Tensor& index,
                                        const at::Tensor& src,
-                                       c10::string_view reduce) {
+                                       c10::string_view reduce,
+                                       bool unique_indices) {
   XLA_FN_COUNTER("xla::");
+  XLA_CHECK(unique_indices)
+      << "scatter not have a deterministic implementation";
   return scatter_reduce_helper(self, dim, index, src, reduce);
 }
 
 at::Tensor XLANativeFunctions::scatter(const at::Tensor& self, int64_t dim,
                                        const at::Tensor& index,
                                        const at::Scalar& value,
-                                       c10::string_view reduce) {
+                                       c10::string_view reduce,
+                                       bool unique_indices) {
```

Comment:
Do we also need to update our c++ test to pass `unique_indices=true`?

Reply:
Yep, updated in the new commit.